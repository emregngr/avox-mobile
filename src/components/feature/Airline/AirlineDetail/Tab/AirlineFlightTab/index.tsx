import React, { useMemo } from 'react'
import { View } from 'react-native'

import { Alliance } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/Alliance'
import { DestinationList } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/DestinationList'
import { RoutesList } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/RoutesList'
import { StatsGrid } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/StatsGrid'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { Airline } from '@/types/feature/airline'

interface AirlineFlightTabProps {
  airlineData: Airline
}

export const AirlineFlightTab = ({ airlineData }: AirlineFlightTabProps) => {
  const { selectedTheme } = useThemeStore()
  const { selectedLocale } = useLocaleStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { network, operations } = airlineData ?? {}
  const { alliance } = operations ?? {}
  const {
    destinationCount,
    destinationCountries,
    destinations,
    domesticConnections,
    internationalConnections,
    routes,
  } = network ?? {}

  const memoizedStats = useMemo(
    () => ({
      destinationCount,
      destinationCountries,
      domesticConnections,
      internationalConnections,
    }),
    [destinationCount, destinationCountries, domesticConnections, internationalConnections],
  )

  const iconColor = useMemo(() => colors.onPrimary100, [colors.onPrimary100])

  const localeStrings = useMemo(
    () => ({
      activeRoutes: getLocale('activeRoutes'),
      alliance: getLocale('alliance'),
      country: getLocale('country'),
      domesticDestinations: getLocale('domesticDestinations'),
      internationalDestination: getLocale('internationalDestination'),
      mainDestinations: getLocale('mainDestinations'),
      totalDestination: getLocale('totalDestination'),
    }),
    [selectedLocale],
  )

  return (
    <View className="px-4 gap-y-4">
      <StatsGrid
        localeStrings={{
          country: localeStrings.country,
          domesticDestinations: localeStrings.domesticDestinations,
          internationalDestination: localeStrings.internationalDestination,
          totalDestination: localeStrings.totalDestination,
        }}
        iconColor={iconColor}
        stats={memoizedStats}
      />

      <DestinationList
        destinations={destinations}
        iconColor={iconColor}
        title={localeStrings.mainDestinations}
      />

      <RoutesList iconColor={iconColor} routes={routes} title={localeStrings.activeRoutes} />

      {alliance !== 'none' && (
        <Alliance alliance={alliance} iconColor={iconColor} title={localeStrings.alliance} />
      )}
    </View>
  )
}
