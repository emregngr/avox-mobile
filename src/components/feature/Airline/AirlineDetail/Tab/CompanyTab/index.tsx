import React from 'react'
import { View } from 'react-native'

import { Company } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Company'
import { Contact } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Contact'
import { Hub } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Hub'
import { Map } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Map'
import { SocialMedia } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/SocialMedia'
import type { Airline } from '@/types/feature/airline'

interface CompanyTabProps {
  airlineData: Airline
}

export const CompanyTab = ({ airlineData }: CompanyTabProps) => {
  const { companyInfo, operations } = airlineData

  return (
    <View className="px-4 gap-y-4">
      <Company airlineData={airlineData} />

      <Contact companyInfo={companyInfo} operations={operations} />

      <SocialMedia companyInfo={companyInfo} />

      <Hub operations={operations} />

      <Map airlineData={airlineData} />
    </View>
  )
}
