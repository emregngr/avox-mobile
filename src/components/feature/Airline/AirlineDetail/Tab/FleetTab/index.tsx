import { router } from 'expo-router'
import React, { useCallback, useMemo } from 'react'
import { View } from 'react-native'

import { FleetHeader } from '@/components/feature/Airline/AirlineDetail/Tab/FleetTab/Sections/FleetHeader'
import { FleetList } from '@/components/feature/Airline/AirlineDetail/Tab/FleetTab/Sections/FleetList'
import { FleetStats } from '@/components/feature/Airline/AirlineDetail/Tab/FleetTab/Sections/FleetStats'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { Airline } from '@/types/feature/airline'

interface FleetTabProps {
  airlineData: Airline
}

export const FleetTab = ({ airlineData }: FleetTabProps) => {
  const { selectedTheme } = useThemeStore()
  const { selectedLocale } = useLocaleStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { fleet, operations } = airlineData || {}

  const { region } = operations || {}

  const { airplaneTypeCount, airplanes, averageAgeYears, totalAirplane } = fleet || {}

  const sortedAirplanes = useMemo(
    () => airplanes?.sort((a, b) => b.count - a.count) || [],
    [airplanes],
  )

  const regionLower = useMemo(() => region?.toLowerCase(), [region])

  const handleImagePress = useCallback(
    (airplaneType: string, imageKey: string) => {
      router.navigate({
        params: {
          regionLower,
          selectedImageKey: imageKey,
          title: airplaneType,
        },
        pathname: '/image-modal',
      })
    },
    [regionLower],
  )

  const localeStrings = useMemo(
    () => ({
      fleetDetail: getLocale('fleetDetail'),
    }),
    [selectedLocale],
  )

  return (
    <View className="px-4 gap-y-4">
      <FleetStats
        airplaneTypeCount={airplaneTypeCount}
        averageAgeYears={averageAgeYears}
        totalAirplane={totalAirplane}
      />

      <FleetHeader colors={colors} fleetDetailText={localeStrings.fleetDetail} />

      <FleetList
        airplanes={sortedAirplanes}
        colors={colors}
        onImagePress={handleImagePress}
        region={region}
        totalAirplane={totalAirplane}
      />
    </View>
  )
}
