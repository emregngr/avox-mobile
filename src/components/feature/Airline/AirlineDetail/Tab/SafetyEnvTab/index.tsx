import React, { useMemo } from 'react'
import { View } from 'react-native'

import { Environmental } from '@/components/feature/Airline/AirlineDetail/Tab/SafetyEnvTab/Sections/Environmental'
import { Safety } from '@/components/feature/Airline/AirlineDetail/Tab/SafetyEnvTab/Sections/Safety'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { AirlineType } from '@/types/feature/airline'

interface SafetyEnvTabProps {
  airlineData: AirlineType
}

export const SafetyEnvTab = ({ airlineData }: SafetyEnvTabProps) => {
  const { selectedLocale } = useLocaleStore()
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { environmental, safety } = airlineData ?? {}
  const { certifications, safetyRecord } = safety ?? {}

  const localeStrings = useMemo(
    () => ({
      certifications: getLocale('certifications'),
      environmentalPolicy: getLocale('environmentalPolicy'),
      environmentalResponsibility: getLocale('environmentalResponsibility'),
      safetyInformation: getLocale('safetyInformation'),
      safetyRecord: getLocale('safetyRecord'),
    }),
    [selectedLocale],
  )

  return (
    <View className="px-4 gap-y-4">
      <Safety
        certifications={certifications}
        certificationsTitle={localeStrings.certifications}
        iconColor={colors.onPrimary100}
        safetyRecord={safetyRecord}
        safetyRecordTitle={localeStrings.safetyRecord}
        title={localeStrings.safetyInformation}
      />

      <Environmental
        content={environmental}
        iconColor={colors.onPrimary100}
        subtitle={localeStrings.environmentalResponsibility}
        title={localeStrings.environmentalPolicy}
      />
    </View>
  )
}
