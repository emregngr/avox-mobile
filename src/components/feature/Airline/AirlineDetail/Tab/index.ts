export * from './AirlineFlightTab'
export * from './CompanyTab'
export * from './FleetTab'
export * from './SafetyEnvTab'
