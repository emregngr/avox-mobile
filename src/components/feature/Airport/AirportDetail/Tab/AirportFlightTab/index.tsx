import { useMemo } from 'react'
import { View } from 'react-native'

import { AirlinesList } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Sections/AirlinesList'
import { RoutesList } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Sections/RoutesList'
import { StatsGrid } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Sections/StatsGrid'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { AirportType } from '@/types/feature/airport'

interface AirportFlightTabProps {
  airportData: AirportType
}

export const AirportFlightTab = ({ airportData }: AirportFlightTabProps) => {
  const { selectedLocale } = useLocaleStore()
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { flightOperations } = airportData ?? {}
  const {
    airlines,
    destinationCount,
    destinationCountries,
    domesticConnections,
    internationalConnections,
    routes,
  } = flightOperations ?? {}

  const memoizedStats = useMemo(
    () => ({
      destinationCount,
      destinationCountries,
      domesticConnections,
      internationalConnections,
    }),
    [destinationCount, destinationCountries, domesticConnections, internationalConnections],
  )

  const localeStrings = useMemo(
    () => ({
      airlines: getLocale('airlines'),
      country: getLocale('country'),
      domesticDestinations: getLocale('domesticDestinations'),
      internationalDestination: getLocale('internationalDestination'),
      popularDestinations: getLocale('popularDestinations'),
      totalDestination: getLocale('totalDestination'),
    }),
    [selectedLocale],
  )

  return (
    <View className="px-4 gap-y-4">
      <StatsGrid
        localeStrings={{
          country: localeStrings.country,
          domesticDestinations: localeStrings.domesticDestinations,
          internationalDestination: localeStrings.internationalDestination,
          totalDestination: localeStrings.totalDestination,
        }}
        iconColor={colors?.onPrimary100}
        stats={memoizedStats}
      />

      <AirlinesList
        airlines={airlines}
        iconColor={colors?.onPrimary100}
        title={localeStrings.airlines}
      />

      <RoutesList
        iconColor={colors?.onPrimary100}
        routes={routes}
        title={localeStrings.popularDestinations}
      />
    </View>
  )
}
