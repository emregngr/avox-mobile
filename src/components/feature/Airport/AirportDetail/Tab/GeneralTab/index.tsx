import React from 'react'
import { View } from 'react-native'

import { Contact } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Contact'
import { General } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/General'
import { Location } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Location'
import { Map } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Map'
import { Services } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Services'
import { SocialMedia } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/SocialMedia'
import type { Airport } from '@/types/feature/airport'

interface GeneralTabProps {
  airportData: Airport
}

export const GeneralTab = ({ airportData }: GeneralTabProps) => {
  const { airportInfo, facilities, operations } = airportData ?? {}

  return (
    <View className="px-4 gap-y-4">
      <General airportInfo={airportInfo} operations={operations} />

      <Services facilities={facilities} />

      <Contact airportInfo={airportInfo} />

      <SocialMedia airportInfo={airportInfo} />

      <Location operations={operations} />

      <Map airportData={airportData} />
    </View>
  )
}
