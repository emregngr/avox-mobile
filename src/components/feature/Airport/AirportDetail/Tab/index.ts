export * from './AirportFlightTab'
export * from './GeneralTab'
export * from './InfrastructureTab'
export * from './NearbyPlacesTab'
