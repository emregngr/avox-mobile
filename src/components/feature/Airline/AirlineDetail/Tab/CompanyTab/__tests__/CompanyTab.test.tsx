import { render } from '@testing-library/react-native'
import React from 'react'

import { CompanyTab } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab'

const mockedCompany = jest.fn()
const mockedContact = jest.fn()
const mockedSocialMedia = jest.fn()
const mockedHub = jest.fn()
const mockedMap = jest.fn()

jest.mock('@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Company', () => {
  const { Text } = require('react-native')
  return {
    Company: (props: any) => {
      mockedCompany(props)
      return <Text testID="company-section">Company Section - {props.airlineData?.name}</Text>
    },
  }
})

jest.mock('@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Contact', () => {
  const { Text } = require('react-native')
  return {
    Contact: (props: any) => {
      mockedContact(props)
      return <Text testID="contact-section">Contact Section - {props.companyInfo?.email}</Text>
    },
  }
})

jest.mock('@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/SocialMedia', () => {
  const { Text } = require('react-native')
  return {
    SocialMedia: (props: any) => {
      mockedSocialMedia(props)
      return <Text testID="social-media-section">Social Media Section</Text>
    },
  }
})

jest.mock('@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Hub', () => {
  const { Text } = require('react-native')
  return {
    Hub: (props: any) => {
      mockedHub(props)
      return <Text testID="hub-section">Hub Section - {props.operations?.hubCount}</Text>
    },
  }
})

jest.mock('@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Map', () => {
  const { Text } = require('react-native')
  return {
    Map: (props: any) => {
      mockedMap(props)
      return <Text testID="map-section">Map Section - {props.airlineData?.icao}</Text>
    },
  }
})

const mockedAirlineData: any = {
  id: '1',
  name: 'Turkish Airlines',
  icao: 'THY',
  iata: 'TK',
  companyInfo: {
    email: 'info@turkishairlines.com',
    phone: '+90 212 444 0 849',
    website: 'https://www.turkishairlines.com',
    address: 'Istanbul, Turkey',
    socialMedia: {
      facebook: 'turkishairlines',
      twitter: 'turkishairlines',
      instagram: 'turkishairlines',
    },
  },
  operations: {
    hubCount: 2,
    destinations: 340,
    fleet: {
      total: 380,
      types: ['A320', 'B737', 'A350', 'B777'],
    },
  },
}

describe('CompanyTab Component', () => {
  it('should render all section components', () => {
    const { getByTestId } = render(<CompanyTab airlineData={mockedAirlineData} />)

    expect(getByTestId('company-section')).toBeTruthy()
    expect(getByTestId('contact-section')).toBeTruthy()
    expect(getByTestId('social-media-section')).toBeTruthy()
    expect(getByTestId('hub-section')).toBeTruthy()
    expect(getByTestId('map-section')).toBeTruthy()
  })

  it('should pass airlineData to Company component', () => {
    render(<CompanyTab airlineData={mockedAirlineData} />)

    expect(mockedCompany).toHaveBeenCalledWith(
      expect.objectContaining({
        airlineData: mockedAirlineData,
      }),
    )
  })

  it('should pass companyInfo to Contact component', () => {
    render(<CompanyTab airlineData={mockedAirlineData} />)

    expect(mockedContact).toHaveBeenCalledWith(
      expect.objectContaining({
        companyInfo: mockedAirlineData.companyInfo,
      }),
    )
  })

  it('should pass companyInfo to SocialMedia component', () => {
    render(<CompanyTab airlineData={mockedAirlineData} />)

    expect(mockedSocialMedia).toHaveBeenCalledWith(
      expect.objectContaining({
        companyInfo: mockedAirlineData.companyInfo,
      }),
    )
  })

  it('should pass operations to Hub component', () => {
    render(<CompanyTab airlineData={mockedAirlineData} />)

    expect(mockedHub).toHaveBeenCalledWith(
      expect.objectContaining({
        operations: mockedAirlineData.operations,
      }),
    )
  })

  it('should pass airlineData to Map component', () => {
    render(<CompanyTab airlineData={mockedAirlineData} />)

    expect(mockedMap).toHaveBeenCalledWith(
      expect.objectContaining({
        airlineData: mockedAirlineData,
      }),
    )
  })

  it('should handle undefined airlineData gracefully', () => {
    render(<CompanyTab airlineData={undefined as any} />)

    expect(mockedContact).toHaveBeenCalledWith(
      expect.objectContaining({
        companyInfo: undefined,
      }),
    )

    expect(mockedHub).toHaveBeenCalledWith(
      expect.objectContaining({
        operations: undefined,
      }),
    )
  })

  it('should handle partial airlineData (missing companyInfo)', () => {
    const partialData = {
      id: '1',
      name: 'Test Airline',
      operations: mockedAirlineData.operations,
    } as any

    render(<CompanyTab airlineData={partialData} />)

    expect(mockedContact).toHaveBeenCalledWith(
      expect.objectContaining({
        companyInfo: undefined,
      }),
    )

    expect(mockedHub).toHaveBeenCalledWith(
      expect.objectContaining({
        operations: partialData.operations,
      }),
    )
  })

  it('should handle partial airlineData (missing operations)', () => {
    const partialData = {
      id: '1',
      name: 'Test Airline',
      companyInfo: mockedAirlineData.companyInfo,
    } as any

    render(<CompanyTab airlineData={partialData} />)

    expect(mockedContact).toHaveBeenCalledWith(
      expect.objectContaining({
        companyInfo: partialData.companyInfo,
      }),
    )

    expect(mockedHub).toHaveBeenCalledWith(
      expect.objectContaining({
        operations: undefined,
      }),
    )
  })

  it('should render with correct container styling', () => {
    const { getByTestId } = render(<CompanyTab airlineData={mockedAirlineData} />)

    const companySection = getByTestId('company-section')
    expect(companySection).toBeTruthy()
  })

  it('should call all section components exactly once', () => {
    render(<CompanyTab airlineData={mockedAirlineData} />)

    expect(mockedCompany).toHaveBeenCalledTimes(1)
    expect(mockedContact).toHaveBeenCalledTimes(1)
    expect(mockedSocialMedia).toHaveBeenCalledTimes(1)
    expect(mockedHub).toHaveBeenCalledTimes(1)
    expect(mockedMap).toHaveBeenCalledTimes(1)
  })
})

describe('CompanyTab Component Snapshot', () => {
  it('should render the CompanyTab Component successfully', () => {
    const { toJSON } = render(<CompanyTab airlineData={mockedAirlineData} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
