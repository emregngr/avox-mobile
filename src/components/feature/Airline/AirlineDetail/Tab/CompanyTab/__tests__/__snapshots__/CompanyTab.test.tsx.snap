// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CompanyTab Component Snapshot should render the CompanyTab Component successfully 1`] = `
<View
  className="px-4 gap-y-4"
>
  <Text
    testID="company-section"
  >
    Company Section - 
    Turkish Airlines
  </Text>
  <Text
    testID="contact-section"
  >
    Contact Section - 
    info@turkishairlines.com
  </Text>
  <Text
    testID="social-media-section"
  >
    Social Media Section
  </Text>
  <Text
    testID="hub-section"
  >
    Hub Section - 
    2
  </Text>
  <Text
    testID="map-section"
  >
    Map Section - 
    THY
  </Text>
</View>
`;
